src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -499,14 +493,18 @@ internal ImmutableArray<IOperation> CreateIgnoredDimensions(BoundNode declaratio
             {
                 case BoundKind.LocalDeclaration:
                     {
-                        return CreateFromArray<BoundExpression, IOperation>(((BoundLocalDeclaration)declaration).DeclaredTypeOpt.BoundDimensionsOpt);
+                        BoundTypeExpression? declaredTypeOpt = ((BoundLocalDeclaration)declaration).DeclaredTypeOpt;
+                        Debug.Assert(declaredTypeOpt != null);
+                        return CreateFromArray<BoundExpression, IOperation>(declaredTypeOpt.BoundDimensionsOpt);
                     }
                 case BoundKind.MultipleLocalDeclarations:
                 case BoundKind.UsingLocalDeclarations:
                     {
                         var declarations = ((BoundMultipleLocalDeclarationsBase)declaration).LocalDeclarations;
+                        BoundTypeExpression? declaredTypeOpt = declarations[0].DeclaredTypeOpt;
```

Comment:
❗ This line accesses `declarations[0]` prior to the `declarations.Length > 0` check